From 04bdb88c26ca1a623904401b8b883354f0f8edf0 Mon Sep 17 00:00:00 2001
From: Roxana Nicolescu <rnicolescu@ciq.com>
Date: Fri, 5 Dec 2025 10:49:16 +0100
Subject: [PATCH] net: Pass NULL to skb_network_protocol() when we don't care
 about vlan depth

jira VULN-66198
cve-pre CVE-2022-49872
commit-author Miaohe Lin <linmiaohe@huawei.com>
commit 2f631133c40cd8e311ae393518c3e651e476ab66

When we don't care about vlan depth, we could pass NULL instead of the
address of a unused local variable to skb_network_protocol() as a param.

	Signed-off-by: Miaohe Lin <linmiaohe@huawei.com>
	Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 2f631133c40cd8e311ae393518c3e651e476ab66)
	Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>
Signed-off-by: Roxana Nicolescu <rnicolescu@ciq.com>
---
 net/core/skbuff.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 7d3922d3fa2d..2cddf045e089 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -3879,7 +3879,6 @@ struct sk_buff *skb_segment(struct sk_buff *head_skb,
 	int err = -ENOMEM;
 	int i = 0;
 	int pos;
-	int dummy;
 
 	if (list_skb && !list_skb->head_frag && skb_headlen(list_skb) &&
 	    (skb_shinfo(head_skb)->gso_type & SKB_GSO_DODGY)) {
@@ -3901,7 +3900,7 @@ struct sk_buff *skb_segment(struct sk_buff *head_skb,
 	}
 
 	__skb_push(head_skb, doffset);
-	proto = skb_network_protocol(head_skb, &dummy);
+	proto = skb_network_protocol(head_skb, NULL);
 	if (unlikely(!proto))
 		return ERR_PTR(-EINVAL);
 
-- 
2.47.3

